### ISSUE TITLE
Add (optional) arbitrary region support with regionmask

### ISSUE THREAD

**Issue by blychs** [NCAR/MELODIES-MONET#272]
Dear all,
If we allow people, as an optional dependency, to include the package "regionmask", we could easily allow for using arbitrary shapefiles, counties, etc.
https://regionmask.readthedocs.io/en/stable/
If you are interested, I should be able to add support for it.
Cheers
Pablo

**Comment by rschwant** [comment]
Yes, sorry this got lost in my emails. Please proceed forward with this. 

We do have filtering options already, so just make sure to consider these filtering options while you are coding this in.
https://melodies-monet.readthedocs.io/en/stable/appendix/yaml.html (search for filter_dict) which can be applied in the observations or in the data_proc section for plotting and statistical calculations.

**Comment by blychs** [comment]
Will be fixed by #326 when merged.

**Comment by rschwant** [comment]
This is done as #326 is merged.
